### PR TITLE
Fix `<BoundConstness as Display>`

### PR DESCRIPTION
**Pull request by Noratrieb**
There was infinite recursion, which is not very good. I'm not sure what the best way to implement this is, I just did something that felt right.

r? @fmease

### CODE REVIEW

File: compiler/rustc_middle/src/ty/mod.rs [hunk 1]
```diff
@@ -333,7 +333,8 @@ impl fmt::Display for BoundConstness {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         match self {
             Self::NotConst => f.write_str("normal"),
```
Comment (compiler-errors) [inline compiler/rustc_middle/src/ty/mod.rs:335]: why are we calling this "normal"

Comment (compiler-errors) [reply compiler/rustc_middle/src/ty/mod.rs:335]: where is this `Display` impl even used? if it's not used, pls remove instead

Comment (Nilstrieb) [reply compiler/rustc_middle/src/ty/mod.rs:335]: It is used, in the error reporting for `TypeError::ConstnessMismatch`.

Comment (compiler-errors) [reply compiler/rustc_middle/src/ty/mod.rs:335]: oh ok

File: compiler/rustc_middle/src/ty/mod.rs [hunk 2]
```diff
@@ -333,7 +333,8 @@ impl fmt::Display for BoundConstness {
...
-            _ => write!(f, "`{self}`"),
```
Comment (fmease) [inline compiler/rustc_middle/src/ty/mod.rs:336]: Oh, that was supposed to be `write!(f, "`{}`", self.as_str())`

**Pull request merged**